### DISCUSSION TITLE
Are there any ways to run mscore in linux cotainer hosted on ARM architecture (M1 Mac)

### DISCUSSION THREAD

**Discussion by ooyamatakehisa** [musescore/MuseScore#20390] [Q&A]
I'm using musescore (AppImage) in centos7 container with XWindow system (xvfb-run command).
Previously I was using intel mac, so I was able to run musescore in the container but once I switched to M1 mac, it doesn't work any more and an error appears:  `/usr/bin/xvfb-run: line 181: /usr/local/bin/mscore: cannot execute binary file`.
It seems like there is not AppImage that can work with ARM architecture.

Are there any ways to achieve this?
Thank you in advance.

**Comment by MarcSabatella** [comment]
There are experimental Linux ARM builds for the upcoming 4.2 release - see the [nightly builds](https://ftp.osuosl.org/pub/musescore-nightlies/linux/4x/nightly/?C=M;O=D) and also the [beta builds](https://ftp.osuosl.org/pub/musescore-nightlies/linux/4x/testing/?C=M;O=D).  No idea if it would actually work within a Linux container on macOS, but it does work in Linux containers on Chromebooks.  Not running M1 chips, though.

**Reply by ooyamatakehisa** [reply]
Thank you for the information. I'll give it a try!

**Reply by ooyamatakehisa** [reply]
Sorry what is "this" in the following message?
> But also see this

**Reply by MarcSabatella** [reply]
Sorry, just a typo leftover from editing, removed now.

**Comment by cbjeukendrup** [comment]
I've tried those arm64 nightly builds in a Parallels Ubuntu machine on my M1 Mac, and that worked well! 

Before these existed, I also tried to run the x86_64 builds using Apple's Rosetta technology; yes, that's actually a macOS thing, but in Parallels Pro (and in self-made virtual machines that use Apple's new Virtualisation framework) there is a way to use it in Linux VMs too. However, I've no luck yet getting MuseScore builds to run with that, because on Ubuntu it's difficult/impossible to install x86_64 versions of the required libraries alongside the arm64 versions of those libraries that are needed by Ubuntu itself. I guess I'll need to use something like Docker maybe. When I have time to experiment again, and I find something, I'll maybe post an update here!

**Comment by ooyamatakehisa** [comment]
Thank you both!

I spent a day trying to use x86_64 AppImage on M1 mac today. As you all probably know, I realized we can still run docker images for x86_64 with Docker Desktop on M1 (AppleSilicon) Mac by specifying `--platform` option ([doc](https://docs.docker.com/build/building/multi-platform/#building-multi-platform-images)). Under the hood, QEMU emulates x86 on arm64 and it should have worked.

But the error persisted and I realized op-mode of CPU (found by the output of `lscpu` command) shows 32 bit. It seems like this is a known issue ([link1](https://github.com/docker/for-mac/issues/6796), [link2](https://forums.docker.com/t/32-bit-op-mode-on-x86-64-linux-container-on-mac-m1/134911/5)) and I had to give up at that point.

I'll try the experimental Linux ARM builds soon and let you all know once I figure out the result!
